src/maxdiffusion/max_utils.py

```diff
@@ -511,6 +515,9 @@ def save_checkpoint(pipeline, params, unet_state, noise_scheduler, config, outpu
   )
   if jax.process_index() == 0:
     walk_and_upload_blobs(config, local_output_dir)
+  
+  # delete files in output_dir to save space
+  shutil.rmtree(local_output_dir)
```

Comment:
Just curious if `save_pretrained` will be performed on all processes and we need to delete it on all processes as well.

Reply:
`save_pretrained` will be saved on all processes, but uploading the weights is only done in the first process. Then we delete the weights from all processes to save memory. This is mainly to ensure we don't run out of disk memory if checkpoint_every is creating a lot of checkpoints.